### ISSUE TITLE
Ranking to prefer longest edges

### ISSUE THREAD

**Issue by jk171505** [dagrejs/dagre-d3#39]
Is there a way to force last nodes position?
![screen shot 2014-04-04 at 11 46 03](https://cloud.githubusercontent.com/assets/1572947/2614167/5ad3394c-bbe2-11e3-860f-9ed218e251a7.png)

**Comment by rexpan** [comment]
@cpettitt It does not work.

**Comment by snowystinger** [comment]
@cpettitt I'm unable to get this rank attribute to work, http://jsfiddle.net/snowystinger/fttup8fb/17/

**Comment by varghesep** [comment]
A lot of people raised the issue of the **rank** not working. Does the author has a paid version of dagre which has **rank** feature?

**Comment by cpettitt** [comment]
No paid version that does this. The work to be done is actually in dagre (vs. dagre-d3). I did a massive refactor to clean up the code and to enable clusters in v0.5.0. I did not port the rank constraints feature mainly because handling sideways edges is complicated. I think porting rank constraints from v0.4.6 should be relatively easy if sideways edges are just ignored. Unfortunately I don't have the time to do the port myself.
